### PR TITLE
feat(ingest): kafka connect metadata ingestion

### PR DESCRIPTION
**Pull request by taufiqibrahim**
## Checklist
- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable)

## Description

Metadata ingestion for Kafka Connect based on Kafka Connect REST API. Currently limited only for Debezium based source connectors. Sink connector not yet implemented.

Implementation summary:
- Fetching connector from Kafka Connect REST API
- Get connector's topic for each connector name
- Parse topic names based on specific Debezium connector specification
- Generate `DataFlowSnapshot` and `DataJobSnapshot` entities

### CODE REVIEW

File: metadata-ingestion/setup.py [hunk 1]
```diff
@@ -187,6 +187,7 @@ def get_long_description():
             "glue = datahub.ingestion.source.glue:GlueSource",
             "hive = datahub.ingestion.source.hive:HiveSource",
             "kafka = datahub.ingestion.source.kafka:KafkaSource",
+            "kafka-connect = datahub.ingestion.source.kafka_connect:KafkaConnectSource",
```
[the remark this answers is not in the document]
Comment (taufiqibrahim) [reply metadata-ingestion/setup.py:190]: resolved in https://github.com/linkedin/datahub/pull/2516/commits/ca969fffadb2f0cd2ba51f5fa4b0d3ae7a383a07

File: metadata-ingestion/src/datahub/ingestion/source/kafka_connect.py [hunk 2]
```diff
@@ -0,0 +1,283 @@
...
+    return lineages
+
+
+@dataclass
+class KafkaConnectSource(Source):
+    """The class for Kafka Connect source.
+
+    Attributes:
+        config (str): Kafka Connect cluster REST API configurations.
+        report (:obj:`int`, optional): .
```
[the remark this answers is not in the document]
Comment (taufiqibrahim) [reply metadata-ingestion/src/datahub/ingestion/source/kafka_connect.py]: fixed on https://github.com/linkedin/datahub/pull/2516/commits/9e879772a63eb2064d556d181b094593cbce6a3f

**Comment by shirshanka** [comment]
Looks like some lint checks are failing. 

You can fix it via: `./gradlew :metadata-ingestion:lintFix`

**Comment by taufiqibrahim** [comment]
> Looks like some lint checks are failing.
> 
> You can fix it via: `./gradlew :metadata-ingestion:lintFix`

Let me try this command. Thanks @shirshanka

**Review by shirshanka — APPROVED** [review]
Thanks for the contribution @taufiqibrahim!

**Pull request merged**